packages/cozy-client/src/store.spec.js

```diff
@@ -230,6 +230,54 @@ describe('Store', () => {
       ).toBe(FILE_1.label)
     })
 
+    it('should respect the query limit', () => {
+      const query = new Q({
+        doctype: 'io.cozy.todos',
+        limit: 2
+      })
+      let queryResult
+      store.dispatch(initQuery('Q1', query))
+      store.dispatch(initQuery('Q2', query))
+      store.dispatch(
+        receiveQueryResult('Q1', {
+          data: [TODO_1, TODO_2]
+        })
+      )
+      store.dispatch(
+        receiveQueryResult('Q2', {
+          data: [TODO_3, TODO_4]
+        })
+      )
+      queryResult = getQueryFromStore(store, 'Q1')
+      expect(queryResult.data.map(x => x._id)).toEqual(['todo_1', 'todo_2'])
+      queryResult = getQueryFromStore(store, 'Q2')
+      expect(queryResult.data.map(x => x._id)).toEqual(['todo_3', 'todo_4'])
```

Comment:
Q1 and Q2 are the same query `query` right? So I don't understand why the result from the store is not the same.

Maybe we should just have a qdef for Q1 (limit: 2) and an other QDef for Q2 (skip: 2, limit: 2) to be more understandable? 

By the way, why did you use two queries? Why not just one: Q1 limit 2, check the queryResult.length then dispatching new todos (via dispatchMutationResult) and check the result?

Reply:
Right, [2a31fff](https://github.com/cozy/cozy-client/pull/1188/commits/2a31fff42ada54d4d713f53c328095d862118c22) should improve this